plugins/pulp_ostree/plugins/lib.py

```diff
@@ -326,13 +341,36 @@ def import_key(self, path, key_ids):
         :param key_ids: A list of key IDs.
         :type key_ids: list
         """
-        self.repository.open()
+        self.open()
         lib = Lib()
         fp = lib.Gio.File.new_for_path(path)
         in_str = fp.read()
-        imported = self.repository.impl.remote_gpg_import(self.id, in_str, key_ids)
+        imported = self.impl.remote_gpg_import(self.id, in_str, key_ids)
         return imported
 
+    @wrapped
+    def list_refs(self):
+        """
+        Get (remote) repository references.
+
+        :return: list of: Ref
+        :rtype: list
+        :raises LibError:
+        """
+        _list = []
+        lib = Lib()
+        self.open()
+        flags = lib.OSTree.RepoPullFlags.COMMIT_ONLY
+        _, summary = self.impl.remote_list_refs(self.id, None)
+        refs = sorted(summary.keys())
+        self.impl.pull(self.id, refs, flags, None)
```

Comment:
This takes a `GCancellable` argument which is not optional on the RHEL7 versions of the libraries.